### PR TITLE
JS: add query: js/stored-xss

### PR DESCRIPTION
**Pull request by esben-semmle**
This is a revival of #142.

Differences from #142:

-   added new query: js/stored-xss,
-   removed additional sources from js/reflected-xss
-   added models of additional file system libraries

The [Evaluation](<https://git.semmle.com/esben/dist-compare-reports/tree/a53239c49a3b37f2de70e54b8cc6cd00>) shows that performance is as expected: a bit of overhead for a new query.

### CODE REVIEW

**Comment by esben-semmle** [comment]
Rebased to resolve change not conflict

**Comment by esben-semmle** [comment]
(updated expected output for new alert message)

File: javascript/ql/src/Security/CWE-079/StoredXss.qhelp [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+		The following example code writes file names directly to a HTTP
+		response. This leaves the website vulnerable to cross-site scripting,
+		if an attacker can choose the file names on the disk.
+
+	</p>
+	<sample src="examples/StoredXss.js" />
+	<p>
+		Sanitizing the file names prevents the vulnerability:
+	</p>
+	<sample src="examples/StoredXssGood.js" />
```
Comment (mc-semmle) [inline javascript/ql/src/Security/CWE-079/StoredXss.qhelp]: Issue with wrapping on the GOOD example

File: change-notes/1.19/analysis-javascript.md [hunk 2]
```diff
@@ -4,11 +4,15 @@
...
 
-| **Query**                   | **Tags**  | **Purpose**                                                        |
-|-----------------------------|-----------|--------------------------------------------------------------------|
-| *@name of query (Query ID)* | *Tags*    |*Aim of the new query and whether it is enabled by default or not*  |
+| **Query**                                     | **Tags**                                             | **Purpose**                                                                                                                                                                 |
+|-----------------------------------------------|------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Stored cross-site scripting (`js/stored-xss`) | security, external/cwe/cwe-079, external/cwe/cwe-116 | Highlights uncontrolled stored values flowing into HTML content, indicating a violation of [CWE-079](https://cwe.mitre.org/data/definitions/79.html). Results shown on lgtm by default. |
```
Comment (mc-semmle) [inline change-notes/1.19/analysis-javascript.md]: Capitalisation of lgtm -> LGTM

**Comment by mc-semmle** [comment]
@esben-semmle - documentation review completed. 
LGTM, a couple of minor issues.

File: javascript/ql/src/Security/CWE-079/StoredXss.qhelp (hunk 1 above)
Comment (mc-semmle) [reply javascript/ql/src/Security/CWE-079/StoredXss.qhelp]: example-wrapping

**Comment by esben-semmle** [comment]
Thank you @mc-semmle. I have capitalized `lgtm`, and added line breaks to both examples.

**Comment by mc-semmle** [comment]
Thank you @esben-semmle - LGTM 👍

**Pull request merged**